### PR TITLE
Name checks

### PR DESCRIPTION
**Pull request by Rulox**
- Now, the plugin checks if the woocommerce plugin is active before
  activating itself.
- Composer and some constants have been updated with the real Plugin name.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "moxie-leean/LeeanPlugin",
-  "description": "Barebones modular WordPress plugin",
+  "name": "moxie-leean/LeanWoocommerceApi",
```
Comment (mitogh) [inline composer.json]: can you separate your package name by dashes `-` like`moxie-lean/woocomerce-api`

File: leeanp.php [hunk 2]
```diff
@@ -21,4 +22,5 @@
 require_once LEAN_WOOCOMMERCE_API_PLUGIN_DIR . 'vendor/autoload.php';
 
 // Run the plugin setup.
-new PluginSetup();
+$setup = new PluginSetup();
+register_activation_hook( __FILE__, array( $setup, 'maybe_deactivate' ) );
```
Comment (mitogh) [inline leeanp.php]: Can you use the shorter syntax for the `array` (`[]`)

File: PluginSetup.php [hunk 3]
```diff
@@ -37,5 +39,21 @@ public static function maybe_deactivate() {
 			wp_die();
 			exit;
 		}
+
+		if ( ! is_plugin_active( self::WOOCOMMERCE_PATH ) ) {
```
Comment (mitogh) [inline PluginSetup.php]: Can we create a  function in order to don't DRY in both places where the plugin should get deactivated.

File: leeanp.php (hunk 2 above)
Comment (Rulox) [inline leeanp.php]: Yeah sure, I just copied from the wp plugin, I didn't remember that syntax! haha Im on it, also with the others. Thank you!

File: PluginSetup.php (hunk 3 above)
Comment (Rulox) [inline PluginSetup.php]: I don't follow you here, I understand you are talking about some kind of refactoring? Which part exactly? the `wp_kses` part? just create another function and pass all the parameters? Thank you @mitogh

**Pull request merged**